drivers/segger/sysview.c

```diff
@@ -365,63 +255,33 @@ void sched_note_irqhandler(int irq, FAR void *handler, bool enter)
             }
         }
 
-      g_sysview.irq[up_cpu_index()] = 0;
+      g_sysview_driver.irq[up_cpu_index()] = 0;
     }
 }
 #endif
 
 #ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
-void sched_note_syscall_enter(int nr, int argc, ...)
+static void sysview_syscall_enter(FAR struct note_driver_s *drv, int nr,
+                                  int argc, va_list *ap)
 {
   nr -= CONFIG_SYS_RESERVED;
-
-  if (sysview_isenabled_syscall(nr) == 0)
-    {
-      return;
-    }
-
-  /* Set the name marker if the current syscall nr is not active */
-
-  if (NOTE_FILTER_SYSCALLMASK_ISSET(nr, &g_sysview.syscall_marker) == 0)
-    {
-      /* Set the name marker */
-
-      SEGGER_SYSVIEW_NameMarker(nr, g_funcnames[nr]);
-
-      /* Mark the syscall active */
-
-      NOTE_FILTER_SYSCALLMASK_SET(nr, &g_sysview.syscall_marker);
-
-      /* Use the Syscall "0" to identify whether the syscall is enabled,
-       * if the host tool is closed abnormally, use this bit to clear
-       * the active set.
-       */
-
-      if (NOTE_FILTER_SYSCALLMASK_ISSET(0, &g_sysview.syscall_marker) == 0)
-        {
-          NOTE_FILTER_SYSCALLMASK_SET(0, &g_sysview.syscall_marker);
-        }
-    }
-
+  SEGGER_SYSVIEW_NameMarker(nr, g_funcnames[nr]);
```

Comment:
it's inefficient to send the name marker every time, why remove the original logic which avoid send the name more than once.

Reply:
done